pal/src/map/virtual.cpp

```diff
@@ -613,6 +631,11 @@ static BOOL VIRTUALReleaseMemory( PCMI pMemoryToBeReleased )
         }
     }
 
+    if (pVirtualMemoryLastFound->startBoundary >= pMemoryToBeReleased->startBoundary)
```

Comment:
Why >=? Should this not be == instead?

Reply:
== condition is always true. Consumers of this method calls for lookup hence we setup the pVirtualMemoryLastFound first. I do check for `>` condition because of calls to this interface from `error` handling of PAL's virtual memory handling. Actually we should discard the value either it is equal or not. However, no point discarding it in case if it's `<`. So, the cost of `==` and `<=` is almost the same here but we might save the cache in case the problem happened in new memory request.

Reply:
Hmm, again, I didn't completely understand- sorry! The case I'm imagining is this:
```pVirtualMemory = 1000 -> 2000 -> 3000
pVirtualMemoryLastFound = 1000
```

Now you call `VIRTUALReleaseMemory` with 1000. In your case, wouldn't you invalidate the cache unnecessarily?